features/FEATURE_BLE/ble/SecurityManager.h

```diff
@@ -14,16 +14,150 @@
  * limitations under the License.
  */
 
-#ifndef __SECURITY_MANAGER_H__
-#define __SECURITY_MANAGER_H__
+#ifndef _SECURITY_MANAGER_H_
+#define _SECURITY_MANAGER_H_
 
 #include <stdint.h>
 
 #include "Gap.h"
 #include "CallChainOfFunctionPointersWithContext.h"
+#include "ble/BLETypes.h"
+
+/**
+ * Overview
+ *
+ * Security Manager is used to provide link security through encryption, signing and authentication
+ * which are made possible by pairing and optionally bonding. Pairing is the process of establishing
+ * and/or exchanging keys used for the current connection. Bonding means saving this information so that
+ * it can later be used after reconnecting without having to pair again. This saves time and power.
+ *
+ * There are many ways to provide these at different levels of security depending on your requirements
+ * and the facilities provided by the application. The process starts with initialising the SecurityManager
+ * with default options for new connections. Some settings can later be changed per link or globally.
+ *
+ * The important settings in the init() function are the MITM requirement and IO capabilities. Man in the
+ * Middle (MITM) protection prevents an attack where one device can impersonate another device by
+ * pairing with both devices at the same time. This protection is achieved by sharing some information
+ * between the devices through some independent channel. The IO capabilities of both devices dictate
+ * what algorithm is used. For details @see BLUETOOTH SPECIFICATION Version 5.0 | Vol 3, Part H - 2.3.5.1.
+ * You can change the IO capabilities after initialisation with setIoCapability(). This will take effect
+ * for all subsequent pairings.
+ *
+ * Sharing this information through IO capabilities means user interaction which limits the degree of
+ * protection due to the limit of the amount of data that we can expect the user to transfer. Another
+ * solution is using OOB (out of band) communication to transfer this data instead which can send much
+ * more data making MITM attack even less likely to succeed. OOB data has to be exchanged by the application
+ * and provided to the Security Manager. Use setOOBDataUsage() to indicate you want to use it. The same call also
+ * allows you to set whether or not the communication channel you are using to transmit the OOB data is
+ * itself secure against MITM protection - this will set the level of the link security achieved using pairing
+ * that uses this data.
+ *
+ * The most secure pairing is provided by Secure Connections which relies on Elliptical Curve Cryptography.
+ * Support for Secure Connections is dependent on both the stack and controller on both sides supporting
+ * it. If either side doesn't support it Legacy Pairing will be used. This is an older standard of pairing.
+ * If higher security is required legacy pairing can be disabled by calling allowLegacyPairing(false);
+ *
+ * How to use
```

Comment:
It maybe interresting to use the tag `par`: https://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdpar

Reply:
added paragraphs